handler/src/main/java/io/netty/handler/ssl/ReferenceCountedOpenSslEngine.java

```diff
@@ -1109,7 +1120,8 @@ public final synchronized void closeOutbound() {
 
     @Override
     public final synchronized boolean isOutboundDone() {
-        return isOutboundDone;
+        // check if there is anything left in the outbound buffer.
+        return outboundClosed && (isDestroyed() || SSL.pendingWrittenBytesInBIO(networkBIO) == 0);
```

Comment:
but `isDestroyed() == true` means that `outboundClosed == true`, and if this is not the case then we have a bug. Am I missing something?